web/frontend/src/lib/schedule.ts

```diff
@@ -0,0 +1,28 @@
+export const scheduleAsyncPoll = (callback: () => Promise<void>, interval: number) => {
+  let cancelled = false;
+  let timeoutId = -1;
+
+  const refreshAndScheduleNext = async () => {
+    await callback();
+
+    // eslint-disable-next-line no-use-before-define
+    scheduleNext();
+  };
+
+  const scheduleNext = () => {
+    if (cancelled) {
+      return;
+    }
+
+    timeoutId = window.setTimeout(refreshAndScheduleNext, interval);
```

Comment:
Maybe rather than an early return, purely stylistic

```suggestion
    if (!cancelled) {
      timeoutId = window.setTimeout(refreshAndScheduleNext, interval);
    }
```